qcodes/tests/dataset/test_dataset_basic.py

```diff
@@ -325,11 +325,25 @@ def test_database_upgrade(empty_temp_db):
     connection = connect(qc.config["core"]["db_location"],
                  qc.config["core"]["db_debug"])
     userversion = get_user_version(connection)
-    if userversion != 0:
+    if userversion != 1:
         raise RuntimeError("trying to upgrade from version 0"
                            " but your database is version"
                            " {}".format(userversion))
     sql = 'ALTER TABLE "runs" ADD COLUMN "quality"'
 
     atomicTransaction(connection, sql)
-    set_user_version(connection, 1)
+    set_user_version(connection, userversion + 1)
+    upgraded_userversion = get_user_version(connection)
+    assert upgraded_userversion == userversion + 1
+
+
+def test_perform_actual_upgrade_0_to_1(dataset):
+    connection = connect(qc.config["core"]["db_location"],
+                 qc.config["core"]["db_debug"])
+    userversion = get_user_version(connection)
+    set_user_version(connection, 0)
```

Comment:
This line and the next effectively performs the downgrade 1 to 0. Would it make sense to provide dedicated downgrading functions in `sqlite_base.py`?

Reply:
I don't think so. I am not sure how to do this best. We don't want users to be able to run with the old version of the db so we unconditionally want to upgrade. On the other hand we want to have the old db so we can test it